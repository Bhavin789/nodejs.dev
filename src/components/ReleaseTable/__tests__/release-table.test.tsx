/* eslint-disable @typescript-eslint/explicit-function-return-type */
import React from 'react';
import renderer from 'react-test-renderer';
import ReleaseTable from '..';

describe('ReleaseTable component', () => {
  it('renders correctly with data', () => {
    const mockReleaseData = [
      {
        version: 'v14.13.0',
        date: '2020-09-29',
        files: ['aix-ppc64', 'headers', 'linux-arm64', 'linux-armv7l'],
        npm: '6.14.8',
        v8: '8.4.371.19',
        uv: '1.40.0',
        zlib: '1.2.11',
        openssl: '1.1.1g',
        modules: '83',
        lts: false,
        security: false,
      },
      {
        version: 'v13.0.1',
        date: '2019-10-23',
        files: ['aix-ppc64', 'headers', 'linux-arm64'],
        npm: '6.12.0',
        v8: '7.8.279.17',
        uv: '1.33.1',
        zlib: '1.2.11',
        openssl: '1.1.1d',
        modules: '79',
        lts: false,
        security: false,
      },
    ];
    const tree = renderer
      .create(<ReleaseTable releases={mockReleaseData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
