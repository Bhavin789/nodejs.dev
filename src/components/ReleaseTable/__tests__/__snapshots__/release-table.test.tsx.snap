// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseTable component renders correctly with data 1`] = `
<div
  className="overflow-horizontal-scroll"
>
  <table>
    <thead>
      <tr>
        <th>
          Version
        </th>
        <th>
          LTS
        </th>
        <th>
          Date
        </th>
        <th>
          V8
        </th>
        <th>
          NPM
        </th>
        <th>
          ABI
        </th>
        <th>
          SHASUM
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          v14.13.0
        </td>
        <td>
          
        </td>
        <td>
          2020-09-29
        </td>
        <td>
          8.4.371.19
        </td>
        <td>
          6.14.8
        </td>
        <td>
          ABI?
        </td>
        <td>
          <a
            href="https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V14.md#14.13.0"
          >
            Changelog
          </a>
        </td>
        <td>
          <a
            href="https://nodejs.org/download/release/v14.13.0/"
          >
            Download
          </a>
        </td>
      </tr>
      <tr>
        <td>
          v13.0.1
        </td>
        <td>
          
        </td>
        <td>
          2019-10-23
        </td>
        <td>
          7.8.279.17
        </td>
        <td>
          6.12.0
        </td>
        <td>
          ABI?
        </td>
        <td>
          <a
            href="https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V13.md#13.0.1"
          >
            Changelog
          </a>
        </td>
        <td>
          <a
            href="https://nodejs.org/download/release/v13.0.1/"
          >
            Download
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
